File: hash/getHash.py
I64BIT_TABLE ='ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789_-'.split();
I64BIT_TABLE_str = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789_-'

def hash(input):
	hsh_ = 5381;
	len_i = len(input) -1;
	if isinstance(input,str):
		for i in range(len_i,-1,-1):
			hsh_ += (hsh_<<5) 
			hsh_ += ord(input[i]);
	else:
		for i in range(len_i,-1,-1):
			hsh_ += (hsh_ << 5) + input[i];
	value = hsh_ & 0x7FFFFFFF;
	#print(value&0x3F)
	retValue = '';
	retValue += I64BIT_TABLE_str[value&0x3F];
	while value > 0:
		retValue += I64BIT_TABLE_str[value&0x3F];
		value = value >> 6
	return retValue[1:]

File: hash/getHash.js
var program = require('commander');

var I64BIT_TABLE =
 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789_-'.split('');
function exec(flag){
 var input = flag;
 var hash = 5381;
 var i = input.length - 1;
 if(typeof input == 'string'){
  for (; i > -1; i--)
   hash += (hash << 5) + input.charCodeAt(i);
 }
 else{
  for (; i > -1; i--)
   hash += (hash << 5) + input[i];
 }
 var value = hash & 0x7FFFFFFF;
  
 var retValue = '';
 do{
  retValue += I64BIT_TABLE[value & 0x3F];
 }
 while(value >>= 6);  
 //return retValue;
 console.log(retValue);
}

program
    .command('exec <flag>')
    .action(function(flag){
        exec(flag);
    });
program.parse(process.argv);